accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -263,6 +270,113 @@ The **flatten** function is modified as follows:
 and for all `R`, if `T <: Future<R>` then `S <: R`; then **flatten**(`T`) = `S`
   - otherwise **flatten**(`T`) = `T`
 
+### The future value type of an asynchronous non-generator function
+
+_We specify a concept which corresponds to the static type of objects which may
+be contained in the Future object returned by an async function with a given
+declared return type._
+
+Let _f_ be an asynchronous non-generator function with declared return type
+`T`. Then the **future value type** of _f_ is **futureValueType**(`T`).
+The function **futureValueType** is defined as follows:
+
+- **futureValueType**(`S?`) = **futureValueType**(`S`), for all `S`.
+- **futureValueType**(`S*`) = **futureValueType**(`S`), for all `S`.
+- **futureValueType**(`Future<S>`) = `S`, for all `S`.
+- **futureValueType**(`FutureOr<S>`) = `S`, for all `S`.
+- **futureValueType**(`void`) = `void`.
+- Otherwise, for all `S`, **futureValueType**(`S`) = `Object?`.
+
+_Note that it is a compile-time error unless the return type of an asynchronous
+non-generator function is a supertype of `Future<Never>`, which means that
+the last case will only be applied when `S` is a non-`void` top type._
```

Comment:
... or `Object`.

(The one non-top-type supertype of Future that we now have)

Reply:
Done.